lib/fog/libvirt/models/compute/server.rb

```diff
@@ -97,6 +97,11 @@ def destroy(options={ :destroy_volumes => false, :flags => ::Libvirt::Domain::UN
           if flags.zero?
             service.vm_action(uuid, :undefine)
           else
+            # the test driver doesn't support UNDEFINE_NVRAM and we can't probe
```

Comment:
Should we have a "FIXME" comment so that it gets highlighted?
Is it possible to open a issue for the test driver so that the UNDEFINE_NVRAM gets fixed soonish?

Reply:
Some more digging in the source tells me it's implemented in [src/test](https://gitlab.com/libvirt/libvirt/-/tree/master/src/test?ref_type=heads). Specifically, https://gitlab.com/libvirt/libvirt/-/blob/239669049d9904e5e8da2d8b2a38d4d927a167e9/src/test/test_driver.c?page=5#L4578-4634

I'm not sure if it can simply be added to `virCheckFlags` there, or if the whole driver needs to support nvram.

Reply:
this is true. The https://gitlab.com/libvirt/libvirt/-/blob/master/src/internal.h?ref_type=heads#L259 is a just a macro and will return if a unsupported flag is used. And https://gitlab.com/libvirt/libvirt/-/blob/239669049d9904e5e8da2d8b2a38d4d927a167e9/src/test/test_driver.c?page=5#L4587 only specified some flags but not NVRAM.